rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,2296 @@
+of the file and the most recently saved version:
+
+```shell
+$ git diff
+```
+
+```text
+diff --git a/mars.txt b/mars.txt
+index df0654a..315bf3a 100644
+--- a/mars.txt
++++ b/mars.txt
+@@ -1 +1,2 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
+```
+
+The output is cryptic because
+it is actually a series of commands for tools like editors and `patch`
+telling them how to reconstruct one file given the other.
+If we break it down into pieces:
+
+1.  The first line tells us that Git is producing output similar to the Unix `diff` command
+    comparing the old and new versions of the file.
+2.  The second line tells exactly which versions of the file
+    Git is comparing;
+    `df0654a` and `315bf3a` are unique computer-generated labels for those versions.
+3.  The third and fourth lines once again show the name of the file being changed.
+4.  The remaining lines are the most interesting, they show us the actual differences
+    and the lines on which they occur.
+    In particular,
+    the `+` marker in the first column shows where we added a line.
+
+After reviewing our change, it's time to commit it:
+
+```shell
+$ git commit -m "Add concerns about effects of Mars' moons on Wolfman"
+$ git status
+```
+
+```text
+On branch master
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git checkout -- <file>..." to discard changes in working directory)
+
+	modified:   mars.txt
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+Whoops:
+Git won't commit because we didn't use `git add` first.
+Let's fix that:
+
+```shell
+$ git add mars.txt
+$ git commit -m "Add concerns about effects of Mars' moons on Wolfman"
+```
+
+```text
+[master 34961b1] Add concerns about effects of Mars' moons on Wolfman
+ 1 file changed, 1 insertion(+)
+```
+
+> **Ignoring things**
+>
+> We're only working with one file so far (`mars.txt`),
+> but what if we had lots of files in this directory?
+> We might want to have git track our text (`.txt`) files, for instance,
+> but not our data files (`.dat`).
+>
+> So that those data files don't show up as "untracked files" every time we call `git status`,
+> we can create a file in the root directory of our project called `.gitignore`.
+> In this file we can list all the files we want git to ignore.
+> For example, we could add the line `*.dat` to `.gitignore` to ignore all the data files.
+> 
+> Don't forget to add and commit the `.gitignore` file to your repository so that git
+> knows to use it!
+
+
+## Remotes in GitHub {#rse-cmdline-git-remotes}
+
+One of the first things we did when learning Git via GitKraken or R Studio
+was to link our local repository with an external hosting service (GitHub).
+We will do that again here,
+so login to GitHub and create a new repository called `planets`.
+
+The next step is to connect the two repositories.
```

Comment:
This is a perfect chance for them to make the repo for the package, so later things like CI and unit testing can build off of this section of the course.